index.md

```diff
@@ -1,213 +1,297 @@
+        </div>
+    </section>
+
+    <section class="homepage__beautiful_uis">
+        <h1>Expressive, beautiful UIs</h1>
+
+        <p>
+        Delight your users with Flutter's built-in
+        beautiful Material Design and
+        Cupertino (iOS-flavor) widgets, rich motion APIs,
+        smooth natural scrolling, and platform awareness.
+        </p>
+
+        <div class="screenshot-list">
+            <img class="screenshot" src="/images/homepage/screenshot-1.png" width="270" height="480" alt="Brand-first shopping design">
+            <img class="screenshot" src="/images/homepage/screenshot-2.png" width="270" height="480" alt="Fitness app design">
+            <img class="screenshot" src="/images/homepage/screenshot-3.png" width="270" height="480" alt="Contact app design">
+            <img class="screenshot" src="/images/homepage/ios-friendlychat.png" width="270" height="480" alt="iOS chat app design">
+        </div>
+
+        <p>
+        Browse the <a href="/widgets/">widget catalog</a>.
+        </p>
+    </section>
+
+    <section class="homepage__hot_reload">
+        <h1>Modern, reactive framework</h1>
+
+        <p>
+        Easily compose your UI with Flutter's
+        modern functional-reactive framework and
+        rich set of platform, layout, and foundation widgets.
+        Solve your tough UI challenges with
+        powerful and flexible APIs for 2D, animation, gestures,
+        effects, and more.
+        </p>
+
+{% prettify dart %}
+class CounterState extends State<Counter> {
+  int counter = 0;
+
+  void increment() {
+    // Tells the Flutter framework that state has changed,
+    // so the framework can run build() and update the display.
+    setState(() {
+      counter++;
+    });
+  }
+
+  Widget build(BuildContext context) {
+    // This method is rerun every time setState is called.
+    // The Flutter framework has been optimized to make rerunning
+    // build methods fast, so that you can just rebuild anything that
+    // needs updating rather than having to individually change
+    // instances of widgets.
+    return new Row(
+      children: <Widget>[
+        new RaisedButton(
+          onPressed: increment,
+          child: new Text('Increment'),
+        ),
+        new Text('Count: $counter'),
+      ],
+    );
+  }
+}
+{% endprettify %}
+
+        <p>
+        Browse the <a href="/widgets/">widget catalog</a>
+        and learn more about the
+        <a href="/widgets-intro/">functional-reactive framework</a>.
+        </p>
+
+    </section>
+
+    <section class="homepage__interop">
+        <h1>Platform interop and native code</h1>
+
+        <p>
+        Make your app come to life
+        with platform APIs, 3rd party SDKs,
+        and native code.
+        Reuse your existing Java, Swift, and ObjC code.
+        </p>
+
+{% prettify dart %}
+// From https://github.com/flutter/flutter/tree/master/examples/platform_channel
```

Comment:
Maybe move the 'From' link outside the code section, and make it a link to the sample.